### PR TITLE
Create common compat library

### PR DESCRIPTION
**Pull request by feliwir**
Avoid files being compiled twice and reduce complexity of cmake build. However we still need to avoid different build flags being used inside those files (`NO_ERR`) in particular

### CODE REVIEW

File: Projects/CMakeLists.txt [hunk 1]
```diff
@@ -262,14 +258,8 @@ IF(LINUX)
     FIND_PACKAGE(OpenAL REQUIRED)
 ENDIF()
 
-#Build static libraries only for FFTW
+#Build static libraries only for subprojects
 SET(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
-#Enable floating point for FFTW
-SET(ENABLE_FLOAT ON CACHE BOOL "" FORCE)
```
Comment (autious) [inline Projects/CMakeLists.txt:268]: Does removing these not have an impact?

**Review by autious — APPROVED** [review]
Maybe OgLogging could be named OvergrowthLogging?

Otherwise it looks good!

File: Projects/CMakeLists.txt (hunk 1 above)
Comment (feliwir) [reply Projects/CMakeLists.txt:268]: No. It was used for a now non-existant library (FFTW)

**Comment by feliwir** [comment]
I renamed the targets to `OvergrowthLogging` and `OvergrowthCompat`

**Pull request merged**